### ISSUE TITLE
Publish all changes

### ISSUE THREAD

**Issue by silkstudio** [sanity-io/sanity#2239]
**Is your feature request related to a problem? Please describe.**
I'm using a Sanity hook to trigger a Netlify build when a Sanity doc is updated but have found that if I'm editing multiple docs and then publish them successively, it will trigger multiple builds.

**Describe the solution you'd like**
It would be super helpful if you could edit multiple docs and then 'publish all changes' in one go rather than having to do it for individual docs.

**Describe alternatives you've considered**
For the moment just doing it individually and potentially spacing changes so as not to trigger multiple successive builds.

**Additional context**
Apologies if something like this already exists, I wasn't able to find much from a quick scan through previous requests but am happy to be educated!

**Comment by marijoo** [comment]
The option to debounce triggers or to publish all documents at once would be super helpful – both ecological and economically. ✌️

**Comment by janhoogeveen** [comment]
Additionally, grouping these into versions would be even better It seems like a logical conclusion to grouping changes!

How Prismic does this: https://user-guides.prismic.io/en/articles/778358-what-is-a-release

**Comment by PutziSan** [comment]
Is there a workaround to publish all, e.g. via HTTP API or cli?

**Comment by tobijafischer** [comment]
Grouping multiple changes into releases sounds really convenient.

**Comment by logemann** [comment]
This is a serious drawback currently. Now i am doing 200 publishes which will take me forever

**Comment by grountish** [comment]
this would be awesome

**Comment by AmodeusR** [comment]
Nothing yet? It gets really tedious to publish each change at a time.

**Comment by miked-known** [comment]
Any movement on this -- this is a killer/must-have feature ?

**Comment by ozymandias547** [comment]
This is a necessity.

**Comment by sitek94** [comment]
+1 it would be really nice to have such feature, especially when making atomic changes to multiple documents

**Comment by olivierolivier** [comment]
+1 Makes a lot of sense both in terms of usability for editors and  to decrease the carbon footprint of a website/app.

**Comment by JayBox325** [comment]
I built a platform that had tags. Now I'm building the tags pages to display content associated with that tag, however, I've got a shed-load of tags with a required slug field that I now have to manually go and update. For the future, it would be great for a document section "mass actions" option, to mass delete, mass publish, mass unpublish etc.

**Comment by kitsbits** [comment]
This is a very common-sense and needed feature. Leaving a comment to make sure this thread stays open. Please do this!

**Comment by artmen1516** [comment]
This is a really important feature, please implement it, could be responsible to change to another CMS that has this...

**Comment by droplab** [comment]
Agreed, this is a crucial feature.  Is this even on the roadmap? Is there an ETA?

**Comment by achyuta116** [comment]
While testing out a website, it'd be convenient to have this feature to group and publish changes, has this feature gotten any attention?

**Comment by grantimbo** [comment]
Currently in the process of transitioning from Prismic to Sanity, and this is the only feature that's missing so far. Hoping to see this feature implemented on Sanity.

**Comment by Maxframe** [comment]
I would indeed appreciate this time-saving feature

**Comment by farmtopeople** [comment]
+1 from our whole company. We need this feature!

**Comment by alivault** [comment]
+1. Needed this feature yesterday. I had to make changes to a bunch of items and there wasn't a way to batch publish all at once, so I had to publish each one individually and it would trigger a build each time.

**Comment by JayBox325** [comment]
The silence form the Sanity team here, on social media and in the official slack is deafening. If fixing this issue was a major task due to how their content model works or something like that, I'd rather they told us that instead of ignoring it. It's a pretty big part of the platform.

**Comment by michaelcpuckett** [comment]
+1 Need this for triggered pushes

**Comment by Jay-Ferguson** [comment]
I feel like this should be a built in feature for any modern CMS. I thought a VC backed project like Sanity would have had this a long time ago.

**Comment by awbranch** [comment]
I need a this too. Any update?

**Comment by faraazm** [comment]
Seriously? We don't have this yet?

**Comment by kelenakamura** [comment]
so using the netlify deploy plugin manually isn't enough? just don't trigger deploys on publishing docs, only deploy when you manually trigger it via the netlify deploy plugin?

**Comment by marijoo** [comment]
> so using the netlify deploy plugin manually isn't enough? just don't trigger deploys on publishing docs, only deploy when you manually trigger it via the netlify deploy plugin?

You mean the [official plugin](https://github.com/sanity-io/sanity-plugin-dashboard-widget-netlify)? That one wasn’t touched for two years and seems to not even be compatible with Dashboard v4.

**Comment by kelenakamura** [comment]
yes was referencing the comment from Apr 22. wondering if that workflow worked for anyone.

**Comment by marijoo** [comment]
Just found this alternative which seems to be working on the first sight: https://github.com/jclusso/sanity-plugin-netlify

**Comment by kmelve** [comment]
Hi folks! We haven't done a great job being present here on GitHub, but we have a solution for you all in the works: [Content Releases](https://www.sanity.io/blog/introducing-sanity-winter-release-2024#5xtauu). We are working on the public beta as we speak. 

Alternatively, as proposed by some of you, installing or adding an affordance that lets Studio users hit a build trigger can also work.

**Comment by kmelve** [comment]
We have now launched [Content Releases](https://www.sanity.io/blog/introducing-content-releases)!

The state of a release is also kept in a document, which [lets you trigger a webhook](https://www.sanity.io/docs/content-releases-cheat-sheet#99b667300c03) when a release goes out, for example to trigger a site rebuild.
